### PR TITLE
Add support for the `#[WithMonologChannel]` attribute of Monolog 3.5 to autoconfigure the channel

### PR DESCRIPTION
**Pull request by stof**
Configuring custom channels on some services is one of the last reasons I have for defining services explicitly in my config files. This attribute allows to solve that.

### CODE REVIEW

File: Attribute/WithLoggerChannel.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+namespace Symfony\Bundle\MonologBundle\Attribute;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class WithLoggerChannel
+{
+    private string $channel;
+
+    public function __construct(string $channel)
+    {
+        $this->channel = $channel;
```
Comment (OskarStark) [inline Attribute/WithLoggerChannel.php:21]: We can use CPP here, can we?

Comment (OskarStark) [reply Attribute/WithLoggerChannel.php:21]: Oh no, we still support Symfony 5.4

Comment (stof) [reply Attribute/WithLoggerChannel.php:21]: For a package supporting PHP 7.2+ ?

File: Attribute/WithLoggerChannel.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Autoconfigures the logger channel used when inject a logger in the service.
+ *
+ * This adds the `monolog.logger` tag on the service definition.
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class WithLoggerChannel
+{
+    private string $channel;
```
Comment (nicolas-grekas) [inline Attribute/WithLoggerChannel.php:22]: what about using a public property marked as `@readonly`?

Comment (stof) [reply Attribute/WithLoggerChannel.php]: Done.

I also made the class final as there is no reason to extend it.

File: Attribute/WithLoggerChannel.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * This adds the `monolog.logger` tag on the service definition.
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
+final class WithLoggerChannel
+{
+    /**
+     * @readonly
+     */
+    public string $channel;
```
Comment (nicolas-grekas) [inline Attribute/WithLoggerChannel.php:25]:
```suggestion
    public $channel;
```

Comment (stof) [reply Attribute/WithLoggerChannel.php:25]: IMO, it is fine to keep the typed property there. Attributes don't work on PHP 7.2 anyway

**Review by lyrixx — APPROVED** [review]
👍🏼 I like it

File: Attribute/WithLoggerChannel.php (hunk 3 above)
Comment (stof) [reply Attribute/WithLoggerChannel.php:25]: ah indeed. For some reasons, I thought CPP was 8.1 like the readonly properties.

**Comment by nicolas-grekas** [comment]
> should we move the attribute to the bridge?

Or better: to monolog itself?

**Comment by stof** [comment]
@Seldaek do you think this attribute should go in Monolog itself ?

**Comment by stof** [comment]
I submitted it to Monolog in https://github.com/Seldaek/monolog/pull/1847
If this is accepted, I will update this PR to use the Monolog attribute instead.

**Pull request merged**